test.js

```diff
@@ -199,9 +234,38 @@ describe('capture-exit', function() {
       });
     });
 
+    it('throws an exit code of the first registered handler which rejects', function() {
```

Comment:
I don't know about exit code preservation being tested. It seems to be that it important to test the following:

all handlers (even the very slow) are invoked before the real `process.exit` is called